### PR TITLE
Add tf2.7 training sagemaker dockerfiles

### PR DESCRIPTION
**Pull request by hsl89**
Add sagemaker dockerfiles for tf2.7

### CODE REVIEW

File: tensorflow/buildspec.yml [hunk 1]
```diff
@@ -84,62 +84,93 @@ images:
...
-  #   docker_file: !join [ docker/, *SHORT_VERSION, /, *DOCKER_PYTHON_VERSION, /Dockerfile.sagemaker.,
-  #                        *DEVICE_TYPE ]
-  #   context:
-  #     <<: *TRAINING_CONTEXT
-  BuildE3TensorflowCPUInferencePy3DockerImage:
-    <<: *INFERENCE_REPOSITORY
-    build: &TENSORFLOW_CPU_INFERENCE_PY3 false
-    image_size_baseline: 4899
-    device_type: &DEVICE_TYPE cpu
+  BuildTensorflowSageMakerGpuPy38Cu112TrainingDockerImage:
```
Comment (tejaschumbalkar) [inline tensorflow/buildspec.yml:87]: duplicate entry, please remove this item

File: tensorflow/training/docker/2.7/py3/cu112/Dockerfile.sagemaker.gpu [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+RUN $PYTHON -m pip install --no-cache-dir -U \
+    "bokeh>=2.3,<3" \
+    "imageio>=2.9,<3" \
+    "opencv-python>=4.3,<5" \
+    "plotly>=5.1,<6" \
+    "seaborn>=0.11,<1" \
+    "numba<0.54" \
+    "shap>=0.39,<1"
+
```
Comment (roywei) [inline tensorflow/training/docker/2.7/py3/cu112/Dockerfile.sagemaker.gpu:26]: Boost is missing:
https://github.com/aws/deep-learning-containers/blob/master/tensorflow/training/docker/2.6/py3/cu112/Dockerfile.gpu#L135

Comment (roywei) [reply tensorflow/training/docker/2.7/py3/cu112/Dockerfile.sagemaker.gpu:26]: You can check your TF compile flag using the following way and decide which flag to use
```
wget https://sourceforge.net/projects/boost/files/boost/1.73.0/boost_1_73_0.tar.gz/download -O boost_1_73_0.tar.gz
tar -xzf boost_1_73_0.tar.gz
cd boost_1_73_0
./bootstrap.sh
ABI_FLAG=$(python3 -c "import tensorflow as tf; ABI=[x for x in tf.sysconfig.get_compile_flags() if 'CXX11_ABI' in x]; print(' '.join(ABI), end='')")
if [[ "$ABI_FLAG" == "-D_GLIBCXX_USE_CXX11_ABI=0" ]]; then
  ./b2 define=_GLIBCXX_USE_CXX11_ABI=0 threading=multi --prefix=/usr -j 64 cxxflags=-fPIC cflags=-fPIC install || true
else
  ./b2 threading=multi --prefix=/usr -j 64 cxxflags=-fPIC cflags=-fPIC install || true
fi
cd ..
```

File: tensorflow/training/docker/2.7/py3/Dockerfile.e3.cpu [hunk 3]
```diff
@@ -128,7 +128,7 @@ RUN ${PIP} install --no-cache-dir -U \
     "Pillow>=9.0,<10.0" \
     python-dateutil==2.8.1 \
     "pyYAML>=5.4,<5.5" \
-    requests==2.24.0 \
+    requests \
```
Comment (tejaschumbalkar) [inline tensorflow/training/docker/2.7/py3/Dockerfile.e3.cpu:131]: unpinned the `requests` package

**Review by Qingzi-Lan — APPROVED** [review]

**Pull request merged**